api/src/components/publication/controller.ts

```diff
@@ -56,45 +58,138 @@ export const getSeedDataPublications = async (
+        topicIds,
+        type
+    } = event.body;
+    const links: { publicationId: string; versionId: string }[] = [];
+
     try {
-        if (event.body.type !== 'PROBLEM' && event.user.role === 'ORGANISATION') {
+        if (type !== 'PROBLEM' && event.user.role === 'ORGANISATION') {
             return response.json(403, {
                 message: 'Organisational accounts can only create Research Problems.'
             });
         }
 
-        if (
-            event.body.selfDeclaration !== undefined &&
-            event.body.type !== 'PROTOCOL' &&
-            event.body.type !== 'HYPOTHESIS'
-        ) {
+        if (directPublish) {
+            if (event.user.role !== 'ORGANISATION') {
+                return response.json(403, {
+                    message: 'Only organisational accounts can publish directly.'
+                });
+            }
+
+            if (helpers.isEmptyContent(content || '')) {
+                return response.json(400, {
+                    message: 'Content field cannot be empty when publishing directly.'
+                });
+            }
+
+            if (!publicationVersionService.validateConflictOfInterest(event.body)) {
+                return response.json(400, {
+                    message:
+                        'Conflict of interest status must either be false, or true and accompanied by a conflict of interest text value in order to direct publish.'
+                });
+            }
+
+            // The publication must be linked to a topic or publication.
+            // If the organisational account doesn't have a default topic,
+            // topic/publication ID(s) must be provided.
+            if (!event.user.defaultTopicId && !topicIds?.length && !linkedPublicationIds?.length) {
+                return response.json(400, {
+                    message:
+                        'At least one topic ID or linked publication ID must be provided, as your organisation does not have a default topic.'
+                });
+            }
+
+            // All proposed links must be valid.
+            if (linkedPublicationIds?.length) {
+                for (const linkTargetId of linkedPublicationIds) {
+                    const validateLink = await linkService.createLinkValidation(
+                        {
+                            existing: false,
+                            type: type
+                        },
+                        linkTargetId,
+                        event.user.id
+                    );
+
+                    if (!validateLink.valid) {
+                        return response.json(validateLink.details.code, {
+                            message: validateLink.details.message
+                        });
+                    } else {
+                        links.push(validateLink.toPublication);
+                    }
+                }
+            }
+
+            // All proposed topics must exist.
+            if (topicIds?.length) {
```

Comment:
Do you need the if check here? Is this checking the array is non empty or that topicIds is not null/undefined? For non empty, the for..of will just not loop and all will be fine.

Reply:
Can't quite follow what's happening here, can topicIds be null/undefined?

Reply:
I don't think there's a need for the "?.length" part here now you mention it, so have just removed those.

Reply:
:+1: Ok so something about problems and default topics and publications not having topics. I'll take your word there :) Looks better without the `?.length` bit anyway